### ISSUE TITLE
LE Audio: Fatal crash when sending Audio data

### ISSUE THREAD

**Issue by Thalley** [zephyrproject-rtos/zephyr#46346]
**Describe the bug**
The broadcast source and unicast client both fail fatally when attempting to send data on a audio stream. 

**To Reproduce**
Flash and run the broadcast source (as that fails standalone) sample. 

**Expected behavior**
The broadcast source to start sending audio data

**Impact**
Showstopper for LE Audio. The issue seems to persist in both samples (although not failing babblesim). 

**Logs and console output**
```
[00:00:00.591,674] <dbg> bt_iso: hci_le_big_complete: BIG[0] 0x200061e4 completed, status 0
[00:00:00.591,796] <dbg> bt_iso: bt_iso_connected: 0x20005be0
[00:00:00.592,498] <dbg> bt_iso: bt_iso_chan_set_state_debug: chan 0x20006218 iso 0x20005be0 connecting -> connected
[00:00:00.592,559] <dbg> bt_audio_broadcast_source: broadcast_source_iso_connected: stream 0x20006218 ep 0x2000623c
[00:00:00.592,620] <dbg> bt_audio_broadcast_source: broadcast_source_set_ep_state: ep 0x2000623c id 0x00 qos-configured -> streaming
[00:00:00.595,306] <dbg> bt_iso: bt_iso_chan_send: chan 0x20006218 len 40
[00:00:00.595,703] <dbg> bt_iso: bt_iso_chan_send: chan 0x20006218 len 40
[00:00:00.603,057] <dbg> bt_audio_broadcast_source: broadcast_source_iso_sent: stream 0x20006218 ep 0x2000617c
[00:00:00.603,118] <err> os: mem_manage_fault: ***** MPU FAULT *****
[00:00:00.603,179] <err> os: mem_manage_fault:   Instruction Access Violation
[00:00:00.603,271] <err> os: esf_dump: r0/a1:  0x00000000  r1/a2:  0x00000019  r2/a3:  0x00000000
[00:00:00.603,363] <err> os: esf_dump: r3/a4:  0xff4672b5 r12/ip:  0x00000000 r14/lr:  0x0000f9e3
[00:00:00.603,424] <err> os: esf_dump:  xpsr:  0xa1000000
[00:00:00.603,485] <err> os: esf_dump: Faulting instruction address (r15/pc): 0xff4672b4
[00:00:00.603,576] <err> os: z_fatal_error: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
[00:00:00.603,637] <err> os: z_fatal_error: Current thread: 0x20005e68 (unknown)
[00:00:00.917,205] <err> os: k_sys_fatal_error_handler: Halting system

```

**Environment (please complete the following information):**
 - Commit SHA or Version used: c62ee3be4e4f9c268a009bced3b4d969d7ccbe46

**Additional context**
Basically the same application does not fail as a babblesim test (https://github.com/zephyrproject-rtos/zephyr/pull/46204)

The iso broadcast and central iso samples are also fully functioning, so it's likely either some parameter or misconfiguration in the LE Audio stack.
